skrub/_multi_agg_joiner.py

```diff
@@ -0,0 +1,417 @@
+"""
+The MultiAggJoiner extends AggJoiner to multiple auxiliary tables.
+"""
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from skrub import _join_utils
+from skrub._agg_joiner import AggJoiner
+from skrub._dataframe._namespace import is_pandas, is_polars
+from skrub._utils import _is_array_like, atleast_2d_or_none
+
+
+class MultiAggJoiner(BaseEstimator, TransformerMixin):
+    """Extension of the AggJoiner to multiple auxiliary tables.
+
+    Apply numerical and categorical aggregation operations on the `cols`
+    to aggregate, selected by dtypes. See the list of supported operations
+    at the parameter `operations`.
+
+    If `cols` is not provided, `cols` are all columns from `aux_tables`,
+    except `aux_keys`.
+
+    As opposed to AggJoiner, all parameters here must be iterable of str,
```

Comment:
maybe we can be a bit more explicit re. how it differs from the joiner.

As opposed to the AggJoiner, here `aux_tables` is a list of tables, each of which will be joined to the main table. therefore `aux_keys` is now a list of keys, of the same length as `aux_tables`, and each entry in `aux_key` is used to join the corresponding aux table. in the same way, each entry in `cols` is a list of columns to aggregate in the corresponding aux table

Therefore if we have a single table we could either use the AggJoiner
AggJoiner(aux_table, key="ID")

or the MultiAggJoiner
AggJoiner([aux_table], keys=["ID"])

maybe in the examples we can also show how using a multiaggjoiner compares to chaining aggjoiners